homeassistant/components/mqtt/client.py

```diff
@@ -299,6 +305,12 @@ def __init__(self, config: ConfigType) -> None:
         client_key = get_file_path(CONF_CLIENT_KEY, config.get(CONF_CLIENT_KEY))
         client_cert = get_file_path(CONF_CLIENT_CERT, config.get(CONF_CLIENT_CERT))
         tls_insecure = config.get(CONF_TLS_INSECURE)
+        if transport == TRANSPORT_WEBSOCKETS:
+            ws_path = config.get(CONF_WS_PATH)
+            ws_headers = config.get(CONF_WS_HEADERS)
+            self._client.ws_set_options(ws_path, ws_headers)
```

Comment:
Does `mqtt.Client.ws_set_options` accept a `None` `ws_path` and `ws_headers`? If not, I'd suggest failing fast with a `KeyError`:
```suggestion
            ws_path = config[CONF_WS_PATH]
            ws_headers = config[CONF_WS_HEADERS]
            self._client.ws_set_options(ws_path, ws_headers)
```

Reply:
Both can be `None` if `transport` == `tcp`